target/linux/ipq40xx/base-files/lib/upgrade/platform.sh

```diff
@@ -93,3 +93,13 @@ platform_do_upgrade() {
 		;;
 	esac
 }
+
+platform_pre_upgrade() {
+	local board=$(board_name)
+
+	case "$board" in
+	mikrotik,lhgg-60ad)
+		[ -z "$(rootfs_type)" ] && mtd erase firmware
+		;;
+	esac
+}
```

Comment:
Missing newline here, there are also others.
Maybe try to grep them, too (it's a little harder here, since it involves newlines).

An easy option would be to use format-patch and then grep for "No newline at end of file" in those.

Reply:
Yeah, grepping in patches was actually quite fast.
This should be resolved.